pkg/env/env_test.go

```diff
@@ -0,0 +1,21 @@
+package env
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import "testing"
+
+func TestEnvironmentType(t *testing.T) {
+	p := &prod{envType: EnvironmentTypeProduction}
+	out := p.Type()
+
+	matching := EnvironmentTypeDevelopment | EnvironmentTypeProduction
+	if out&matching == 0 {
+		t.Fatal("Doesn't match expected mask")
+	}
+
+	matching = EnvironmentTypeIntegration | EnvironmentTypeDevelopment
+	if out&matching != 0 {
+		t.Fatal("Doesn't match expected mask")
+	}
+}
```

Comment:
This is very clever. I had trouble understanding it at first, so I made an [illustration](https://play.golang.org/p/DvAX-z254Kb). Maybe it will help somebody else, too.

```
EnvironmentTypeProduction:  001
EnvironmentTypeIntegration: 010
EnvironmentTypeDevelopment: 100

matching: EnvironmentTypeDevelopment | EnvironmentTypeProduction
matching:                        100 | 001
matching:                           101
out:                                001
out&matching:                       001

matching: EnvironmentTypeIntegration | EnvironmentTypeDevelopment
matching:                        010 | 100
matching:                           110
out:                                001
out&matching:                       000
```

Reply:
As a side note, the things that use enumer does this as well, but the generated enumer also provides a `.String` method which returns a pretty-printed name and a way of going from string -> enumed value with, eg. `InstallPhaseString("InstallPhaseBootstrap")`. I didn't add that in this PR, since I didn't need all the features that enumer generated.